healsparse/healSparseMap.py

```diff
@@ -262,3 +262,17 @@ def generateHealpixMap(self, nside=None, reduction='mean'):
 
         pass
 
+    def degrade(self, nside_out, reduction='mean'):
+        """
+        Reduce the resolution, i.e., increase the pixel size
+        of a given sparse map
+        """
+        try:
+           assert(self._nsideSparse > nside_out)
+        except ValueError as e:
+           print('nside_out should be smaller than nside for the sparseMap')
+        npop_pix = np.count_nonzero(self.coverageMask)
+        self._sparseMap[self._sparseMap==hp.UNSEEN] = np.nan
+        self._sparseMap = np.nanmean(self._sparseMap.reshape((npop_pix+1, (nside_out//self._nsideCoverage)**2, -1)), axis=2).flatten()
```

Comment:
That's very clever, I didn't know about `np.nanmean` and I bet that takes care of things.  Note that this only works with the `reduction='mean'` option, we should either remove that for the time being, or flag that we will add a (e.g.) median in the future (which will be more expensive of course).

Reply:
So this is going to be have to be updated for the recarray support.  I think that's pretty easy.  If it's a recarray (`self._isRecArray`) then you need to (a) set the sparseMap dtype; (b) loop over fields; but (c) the `UNSEEN` check is on `self._sparseMap[self._primary] == hp.UNSEEN`.
Oh, I just realized that this won't work with the "in-place" replacement, but that's only a couple more lines.